js/src/dom/manipulator.js

```diff
@@ -0,0 +1,56 @@
+import Util from '../util'
+
+/**
+ * --------------------------------------------------------------------------
+ * Bootstrap (v4.0.0-beta): dom/manipulator.js
+ * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
+ * --------------------------------------------------------------------------
+ */
+
+const Manipulator = {
+  setChecked(input, val) {
+    if (input instanceof HTMLInputElement) {
+      input.checked = val
+      input.bsChecked = val
+    }
+  },
+
+  isChecked(input) {
+    if (input instanceof HTMLInputElement) {
+      return input.bsChecked || input.checked
+    }
+    throw new Error('INPUT parameter is not an HTMLInputElement')
+  },
+
+  setDataAttribute(element, key, value) {
+    const $ = Util.jQuery
+    if (typeof $ !== 'undefined') {
+      $(element).data(key, value)
+    }
+
+    element.setAttribute(`data-${key.replace(/[A-Z]/g, (chr) => `-${chr.toLowerCase()}`)}`, value)
+  },
+
+  removeDataAttribute(element, key) {
+    const $ = Util.jQuery
+    if (typeof $ !== 'undefined') {
+      $(element).removeData(key)
+    }
+
+    element.removeAttribute(`data-${key.replace(/[A-Z]/g, (chr) => `-${chr.toLowerCase()}`)}`)
+  },
+
+  toggleClass(element, className) {
+    if (typeof element === 'undefined' || element === null) {
+      return
+    }
+
+    if (element.classList.contains(className)) {
```

Comment:
Would it not be possible to just call `element.classList.toggle(className)`?

Reply:
Is it supported on IE 11?

Reply:
IE (any version) doesn't support the `toggle` method of `Element.classList`: https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#Browser_compatibility

Reply:
According to <https://caniuse.com/#search=classList> it just "does not support the second parameter for the `toggle` method".

Reply:
IE11 appears to have this feature, but IE10 doesn't.

Reply:
@tmorehouse The link you posted has been updated to reflect that IE10 does indeed have basic support for it.